portfolio/src/main/java/com/google/sps/data/RequestJson.java

```diff
@@ -0,0 +1,240 @@
+                                                        // the JSON string with all the data
+    JSONArray allItems = myResponse.getJSONArray("items");
+
+    for (int i = 0; i < numberResults; i++) {
+      returnedList.add(jsonToFullBook(allItems.getJSONObject(i))); // jsonToFullBook
+    }
+    return returnedList;
+  }
+
+  public FullBook jsonToFullBook(JSONObject jsonObject) throws JSONException {
+    FullBook newBook = new FullBook();
+    // Map<String, String> tempISBN = new HashMap<String, String>();
+    String tempISBN = ""; // double nested [{}]
+
+    tempISBN = getIsbn(jsonObject).get("ISBN_13");
+    try {
+      newBook.setStringField("isbn", tempISBN);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+
+    for (BookFieldsEnum field : BookFieldsEnum.values()) {
+      String jsProperty = field.getJSProperty(); // the string value in the enum
+      String tempProperty = "Undefined";
+      List<String> tempPropertyArray = new ArrayList<String>();
+      boolean isArray = false;
+
+      if (!jsProperty.equals("genre") && !jsProperty.equals("isbn")) {
+        try {
+          if (jsProperty.equals("categories") || jsProperty.equals("authors")) {
+            // seperate because they exists as arrays in the JSON object [] and genre has a
+            // different name ("categories")
+            isArray = true;
+            if (jsonObject.getJSONObject("volumeInfo").has(jsProperty)) {
+              tempPropertyArray = jsonArrayToStringArray(
+                  jsonObject.getJSONObject("volumeInfo").getJSONArray(jsProperty));
+            } else {
+              tempPropertyArray = null;
+            }
+          } else if (jsProperty.equals("thumbnail")) {
+            // seperate because it is nested in another object {}
+            if (jsonObject.getJSONObject("volumeInfo").has("imageLinks")
+                && jsonObject.getJSONObject("volumeInfo")
+                       .getJSONObject("imageLinks")
+                       .has(jsProperty)) {
+              tempProperty = jsonObject.getJSONObject("volumeInfo")
+                                 .getJSONObject("imageLinks")
+                                 .get(jsProperty)
+                                 .toString();
+            }
+          } else {
+            if (jsonObject.getJSONObject("volumeInfo").has(jsProperty)) {
+              tempProperty = jsonObject.getJSONObject("volumeInfo").get(jsProperty).toString();
+            }
+          }
+        } catch (JSONException e) {
+          e.printStackTrace();
+        }
+        try {
+          if (isArray) {
+            newBook.setArrayField(jsProperty, tempPropertyArray);
+          } else {
+            newBook.setStringField(jsProperty, tempProperty);
+          }
+        } catch (NoSuchFieldException e) {
+          e.printStackTrace();
+        }
+      }
+    }
+    return newBook;
+  }
+
+  public Map<String, String> getIsbn(JSONObject item) throws JSONException {
```

Comment:
you can also use an ImmutableMap here since I don't think this should ever be modified!